core/mobile-menu.ts

```diff
@@ -0,0 +1,47 @@
+import { atom, computed } from 'nanostores'
+
+import { fastCategories } from './fast.ts'
+import { router } from './router.ts'
+import { slowCategories } from './slow.ts'
+
+const isMenuShowed = atom<boolean>(false)
+
+const isMenuHasItems = computed(
+  [router, fastCategories, slowCategories],
+  (r, f, s) => {
+    if (r.route === 'fast') {
+      return !f.isLoading && f.categories.length > 1
+    }
+
+    if (r.route === 'slow') {
+      return !s.isLoading && s.tree.length > 1
+    }
+
+    return true
+  }
+)
+
+export function toggleMenu(): void {
+  isMenuShowed.set(!isMenuShowed.get())
+}
+
+export const isMenuOpened = computed(
+  [isMenuShowed, isMenuHasItems],
+  (isShowed, isHasItems) => {
+    return isShowed && isHasItems
+  }
+)
+
+router.subscribe((newRoute, prevRoute) => {
+  let targetRoutes = ['add', 'slow', 'fast']
+
+  if (
+    prevRoute &&
+    targetRoutes.includes(newRoute.route) &&
+    newRoute.route !== prevRoute.route
+  ) {
+    setTimeout(() => {
```

Comment:
Why do we have `setTimeout` here?

Reply:
We have a listener for a click to close the menu on any part of the document, this is a micro delay before opening the menu on another route. When I test manually, sometimes the menu does not open when switching routes. But it works with a micro delay.